include/dali/core/convert.h

```diff
@@ -70,6 +70,30 @@ struct needs_clamp {
     (from_unsigned && !to_unsigned && sizeof(To) <= sizeof(From));
 };
 
+// For whatever reason clang tries the templates instead the overloads, so we sfinae the
+// generic clamp out and let it use the explicit overloads for __half
+#if defined(__clang__) && defined(__CUDA__)
+template <typename To>
+struct needs_clamp<__half, To> {
+  static constexpr bool value = false;
+};
+
+template <typename From>
+struct needs_clamp<From, __half> {
+  static constexpr bool value = false;
+};
+
+template <typename T, typename U>
+using skip_template =
```

Comment:
Perhaps use a more specific name for something that's in namesace `dali`:
```suggestion
using skip_clamp_template =
```